src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -79,57 +90,58 @@ public DesignerAttributeDiscoveryService()
             if (!project.SupportsCompilation)
                 return;
 
-            var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-            var designerCategoryType = compilation.DesignerCategoryAttributeType();
-            if (designerCategoryType == null)
-                return;
+            // Defer expensive work until it's actually needed.
+            var lazyProjectVersion = AsyncLazy.Create(project.GetSemanticVersionAsync, cacheResult: true);
```

Comment:
The original code invoked `GetDependentSemanticVersionAsync`. Is the change to `GetSemanticVersionAsync` intentional here?

Reply:
yes.  will add comment.